dnf5/main.cpp

```diff
@@ -933,6 +933,106 @@ static void print_new_leaves(Context & context) {
 }  // namespace dnf5
 
 
+// Recursively search command and its parents whether given argument is configured
+static bool has_named_arg(libdnf5::cli::ArgumentParser::Command * command, std::string_view arg_name) {
+    while (command) {
+        for (const auto & arg : command->get_named_args()) {
+            if (arg->get_long_name() == arg_name) {
+                return true;
+            }
+        }
+        command = command->get_parent() == command ? nullptr : command->get_parent();
+    }
+    return false;
+}
+
+static void print_resolve_hints(dnf5::Context & context) {
+    auto & conf = context.base.get_config();
+    std::vector<std::string> hints;
+    auto transaction_problems = context.get_transaction()->get_problems();
+    auto * command = context.get_selected_command()->get_argument_parser_command();
+
+    // hint --skip-unavailable if a package was not found
+    if ((transaction_problems & libdnf5::GoalProblem::NOT_FOUND) == libdnf5::GoalProblem::NOT_FOUND &&
+        !conf.get_skip_unavailable_option().get_value()) {
+        const std::string_view arg{"--skip-unavailable"};
+        if (has_named_arg(command, arg.substr(2))) {
+            hints.emplace_back(libdnf5::utils::sformat(_("{} to skip unavailable packages"), arg));
+        }
+    }
+
+    if ((transaction_problems & libdnf5::GoalProblem::SOLVER_ERROR) == libdnf5::GoalProblem::SOLVER_ERROR) {
+        bool conflict = false;
+        bool broken_file_dep = false;
+        bool best = false;
+        // walk through all solver problem to detect a conflict, missing file dependency and best
+        for (const auto & resolve_log : context.get_transaction()->get_resolve_logs()) {
+            if (resolve_log.get_problem() == libdnf5::GoalProblem::SOLVER_ERROR) {
+                for (const auto & solv_prob : resolve_log.get_solver_problems()->get_problems()) {
+                    for (const auto & [rule, params] : solv_prob) {
+                        switch (rule) {
+                            case libdnf5::ProblemRules::RULE_PKG_CONFLICTS:
+                                // TODO(mblaha): we should check whether the conflict involves an installed package (missing API).
+                                // https://github.com/rpm-software-management/dnf5/issues/1324
+                                conflict = true;
+                                break;
+                            case libdnf5::ProblemRules::RULE_PKG_NOTHING_PROVIDES_DEP:
+                                if (params.size() >= 1) {
+                                    if (params[0].starts_with('/')) {
+                                        broken_file_dep = true;
+                                    }
+                                }
+                                break;
+                            case libdnf5::ProblemRules::RULE_BEST_1:
+                            case libdnf5::ProblemRules::RULE_BEST_2:
```

Comment:
Optional: I am proposing to add one hint more for `package dnf-data-4.19.0-20240326004737.13.ga6d82221.fc38.noarch is filtered out by exclude filtering`.